bpf/lib/nodeport.h

```diff
@@ -195,7 +195,9 @@ nodeport_fib_lookup_and_redirect(struct __ctx_buff *ctx,
 	__u32 oif;
 	int ret;
 
-	ret = (int)fib_lookup(ctx, &fib_params->l, sizeof(fib_params->l), 0);
+	ret = (int)fib_lookup(ctx, &fib_params->l, sizeof(fib_params->l),
+			      neigh_resolver_available()
+			      ? fib_lookup_skip_neigh() : 0);
```

Comment:
Nit (i.e., not going to block on this): It feels like this would be a bit more readable with a new helper function:
```c
static __always_inline __u64 fib_lookup_neigh_flags(void)
{
        return neigh_resolver_available()? fib_lookup_skip_neigh() : 0;
}
```
which you could then just use in the 4 places to mask the condition. Or you could also fold the condition in `fib_lookup_skip_neigh()` directly.

Reply:
Done. I just rolled this all into `fib_lookup_skip_neigh` to make things simpler.